discovery_engine_core/core/src/storage.rs

```diff
@@ -141,8 +163,28 @@ pub mod models {
         pub(crate) market: Market,
     }
 
+    #[derive(Debug, PartialEq)]
     pub(crate) struct NewscatcherData {
         pub(crate) domain_rank: u64,
         pub(crate) score: Option<f32>,
     }
+
+    #[derive(Debug, PartialEq, Eq)]
+    pub(crate) struct Search {
+        pub(crate) search_by: SearchBy,
+        pub(crate) search_term: String,
```

Comment:
❔ I'm not sure how valid this suggestion will be, but what do you think about wrapping it in `SearchTerm` which upon construction checks if the `String` is not empty? do we need smth like that?

Reply:
we do the search term validation and sanitazation in the engine in the `search_by_query()` and `search_by_topic()` methods. after that we should be safe to assume that the search terms are wellformed.